packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.java

```diff
@@ -307,6 +308,10 @@ private void launchCameraWithPermissionsGranted(Promise promise, Intent cameraIn
     try {
       imageFile = new File(ImagePickerFileUtils.generateOutputPath(mContext.getCacheDir(),
           "ImagePicker", mediaTypes.equals("Videos") ? ".mp4" : ".jpg"));
+      boolean fileExists = !imageFile.createNewFile();
+      if (fileExists) {
+        Log.w(getName(), String.format("File %s already exists.", imageFile.getAbsolutePath()));
+      }
```

Comment:
Ok, so you create this file before launching the camera, right? 
And where is the previous place (line in codebase) this file was created? I couldn't find it 😞 
Is it possible that majority of devices creates the file during taking a picture or something like this? 🤔

Reply:
So, I think the most devices to implement camera intent use streams. Hence, their code should look like this `Stream(File(String))`. In this implementation, we don't need to create a file before.

However, for some reason,  OpePlus always destroyed our activity while the user's taking a photo. As a consequence, the result from intent is lost. I've tried to deduce what can cause this behaviour.  I've looked into [the Android documentation](https://developer.android.com/training/camera/photobasics#TaskPath) and flutter implementation: https://github.com/flutter/plugins/blob/afb5d4dd43c8f631095b4c44017f6136ce78291f/packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java#L371.
It's turned out that they always create the file before they start the camera. So, I do the same. It's helped :D 

Also, I think it's only a temporary fix. We should add methods which will be able to handle activity restart (like this one: https://github.com/flutter/plugins/tree/master/packages/image_picker#handling-mainactivity-destruction-on-android).